src/resources/triton_python_backend_utils.py

```diff
@@ -340,6 +342,23 @@ def set_max_batch_size(self, max_batch_size):
                 str(max_batch_size))
         else:
             self._model_config["max_batch_size"] = max_batch_size
+            if max_batch_size > 1:
+                self.set_dynamic_batching()
```

Comment:
I think it we should not implicitly set the dynamic batching setting to ON when max_batch_size is greater than zero. Since currently there is not way for the user to set max_batch_size to greater than one and not turn on dynamic batching.

Reply:
I think Python backend is a bit different from other backends in that it is more a Python API for the Triton's C-API. Because of this I think the API should be complete and have maximum flexiblity.